### PR TITLE
Review string unions

### PR DESCRIPTION
**Pull request by IlyaKhD**

### CODE REVIEW

File: js/ui/gantt.d.ts [hunk 1]
```diff
@@ -40,15 +40,21 @@ import { Skip } from '../core/index';
...
+import {
+    FirstDayOfWeek,
+} from '../common';
+
+export type GanttContextMenuItem = 'undo' | 'redo' | 'expandAll' | 'collapseAll' | 'addTask' | 'deleteTask' | 'zoomIn' | 'zoomOut' | 'deleteDependency' | 'taskDetails';
+export type GanttPdfExportDateRange = 'all' | 'visible';
+export type GanttPdfExportMode = 'all' | 'treeList' | 'chart';
+export type GanttRenderScaleType = 'minutes' | 'hours' | 'sixHours' | 'days' | 'weeks' | 'months' | 'quarters' | 'years' | 'fiveYears';
+export type GanttScaleType = 'auto' | 'minutes' | 'hours' | 'sixHours' | 'days' | 'weeks' | 'months' | 'quarters' | 'years';
+export type GanttSortingMode = 'multiple' | 'none' | 'single';
```
Comment (AnnaKondratova) [inline js/ui/gantt.d.ts]:
```suggestion

```

File: js/common/grids.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export type GridEnterKeyAction = 'startEdit' | 'moveFocus';
+export type GridEnterKeyDirection = 'none' | 'column' | 'row';
+export type GridFilterOperations = '=' | '<>' | '<' | '<=' | '>' | '>=' | 'contains' | 'endswith' | 'isblank' | 'isnotblank' | 'notcontains' | 'startswith' | 'between' | 'anyof' | 'noneof';
+export type GridGroupingExpandMode = 'buttonClick' | 'rowClick';
+export type GridNewRowPosition = 'first' | 'last' | 'pageBottom' | 'pageTop' | 'viewportBottom' | 'viewportTop';
+export type GridPagerDisplayMode = 'adaptive' | 'compact' | 'full';
+// eslint-disable-next-line @typescript-eslint/no-type-alias
+export type GridPagerPageSize = 'all';
+export type GridRowRenderingMode = 'standard' | 'virtual';
+export type GridSelectionShowCheckBoxesMode = 'always' | 'none' | 'onClick' | 'onLongTap';
```
Comment (AnnaKondratova) [inline js/common/grids.d.ts]:
```suggestion
export type SelectionColumnDisplayMode = 'always' | 'none' | 'onClick' | 'onLongTap';
```

File: js/common/grids.d.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+export type GridApplyFilterMode = 'auto' | 'onClick';
+export type GridColumnButtonName = 'cancel' | 'delete' | 'edit' | 'save' | 'undelete';
```
Comment (AnnaKondratova) [inline js/common/grids.d.ts]:
```suggestion
export type GridColumnButton = 'cancel' | 'delete' | 'edit' | 'save' | 'undelete';
```

File: js/common.d.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ * @public
+ * @namespace DevExpress.common
+ */
+export type ButtonStyle = 'text' | 'outlined' | 'contained';
+
+/**
+ * @public
+ * @namespace DevExpress.common
+ */
+export type SelectionMode = 'multiple' | 'single' | 'none';
```
Comment (AnnaKondratova) [inline js/common.d.ts]:
```suggestion
export type GroupSelectionMode = 'multiple' | 'single' | 'none';
```

File: js/common.d.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * @public
+ * @namespace DevExpress.common
+ */
+export type ForcedSelectionMode = 'multiple' | 'single';
```
Comment (AnnaKondratova) [inline js/common.d.ts]:
```suggestion
export type SelectionMode = 'multiple' | 'single';
```

File: js/ui/list.d.ts [hunk 6]
```diff
@@ -51,6 +50,9 @@ interface ListItemInfo<TItem extends ItemLike> {
     readonly itemIndex: number | { group: number; item: number };
 }
 
+/** @public */
+export type ListSelectionMode = 'all' | 'multiple' | 'none' | 'single';
```
Comment (AnnaKondratova) [inline js/ui/list.d.ts]:
```suggestion
export type CollectionSelectionMode = 'all' | 'multiple' | 'none' | 'single';
```

File: js/animation/position.d.ts [hunk 7]
```diff
@@ -3,12 +3,13 @@ import {
...
-  PositionAlignment,
-  PositionResolveCollisionXY,
-  PositionResolveCollision,
-} from '../types/enums';
+    HorizontalAlignment,
+    PositionAlignment,
+    VerticalAlignment,
+} from '../common';
+
+export type CollisionResolution = 'fit' | 'flip' | 'flipfit' | 'none';
```
Comment (AnnaKondratova) [inline js/animation/position.d.ts:12]:
```suggestion
export type Collision = 'fit' | 'flip' | 'flipfit' | 'none';
```

File: js/animation/position.d.ts [hunk 8]
```diff
@@ -3,12 +3,13 @@ import {
...
+export type CollisionResolutionCombination = 'fit' | 'fit flip' | 'fit flipfit' | 'fit none' | 'flip' | 'flip fit' | 'flip none' | 'flipfit' | 'flipfit fit' | 'flipfit none' | 'none' | 'none fit' | 'none flip' | 'none flipfit';
```
Comment (AnnaKondratova) [inline js/animation/position.d.ts:14]:
```suggestion
export type CollisionCombination = 'fit' | 'fit flip' | 'fit flipfit' | 'fit none' | 'flip' | 'flip fit' | 'flip none' | 'flipfit' | 'flipfit fit' | 'flipfit none' | 'none' | 'none fit' | 'none flip' | 'none flipfit';
```

File: js/common.d.ts [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+ * @public
+ * @namespace DevExpress.common
+ */
+export type SubmenuShowMode = 'onClick' | 'onHover';
+
+/**
+ * @public
+ * @namespace DevExpress.common
+ */
+export type TextBoxPredefinedButton = 'clear'; // eslint-disable-line @typescript-eslint/no-type-alias
```
Comment (AnnaKondratova) [inline js/common.d.ts:227]:
```suggestion
export type TextBoxButton = 'clear'; // eslint-disable-line @typescript-eslint/no-type-alias
```

File: js/common/charts.d.ts [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type DiscreteAxisDivisionMode = 'betweenLabels' | 'crossLabels';
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type HatchingDirection = 'left' | 'none' | 'right';
```
Comment (AnnaKondratova) [inline js/common/charts.d.ts]:
```suggestion
export type HatchDirection = 'left' | 'none' | 'right';
```

File: js/common/grids.d.ts [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type GridsEditRefreshMode = 'full' | 'reshape' | 'repaint';
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type GroupingExpandMode = 'buttonClick' | 'rowClick';
```
Comment (AnnaKondratova) [inline js/common/grids.d.ts]:
```suggestion
export type GroupExpandMode = 'buttonClick' | 'rowClick';
```

File: js/common/grids.d.ts [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type StartEditAction = 'click' | 'dblClick';
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type StateStoringType = 'custom' | 'localStorage' | 'sessionStorage';
```
Comment (AnnaKondratova) [inline js/common/grids.d.ts]:
```suggestion
export type StateStoreType = 'custom' | 'localStorage' | 'sessionStorage';
```

File: js/common/grids.d.ts [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type ApplyFilterMode = 'auto' | 'onClick';
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type ChangesApplyMode = 'instantly' | 'onDemand';
```
Comment (AnnaKondratova) [inline js/common/grids.d.ts]:
```suggestion
export type ApplyChangesMode = 'instantly' | 'onDemand';
```

File: js/common/charts.d.ts [hunk 14]
```diff
@@ -0,0 +1,190 @@
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type AnimationEasing = 'easeOutCubic' | 'linear';
```
Comment (AnnaKondratova) [inline js/common/charts.d.ts]:
```suggestion
export type AnimationEaseMode = 'easeOutCubic' | 'linear';
```

File: js/common/charts.d.ts [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type AxisScaleType = 'continuous' | 'discrete' | 'logarithmic';
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type ChartsAxisLabelOverlapping = 'rotate' | 'stagger' | 'none' | 'hide';
```
Comment (AnnaKondratova) [inline js/common/charts.d.ts]:
```suggestion
export type ChartsAxisLabelOverlap = 'rotate' | 'stagger' | 'none' | 'hide';
```

File: js/common/charts.d.ts [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type ChartsLabelOverlapping = 'hide' | 'none' | 'stack';
```
Comment (AnnaKondratova) [inline js/common/charts.d.ts]:
```suggestion
export type ChartsLabelOverlap = 'hide' | 'none' | 'stack';
```

File: js/common/charts.d.ts [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type HatchingDirection = 'left' | 'none' | 'right';
+
+/**
+ * @public
+ * @namespace DevExpress.common.charts
+ */
+export type LabelOverlapping = 'hide' | 'none';
```
Comment (AnnaKondratova) [inline js/common/charts.d.ts]:
```suggestion
export type LabelOverlap = 'hide' | 'none';
```

File: js/ui/data_grid.d.ts [hunk 18]
```diff
@@ -80,50 +80,89 @@ import {
...
+    SelectedFilterOperation,
+    SelectionColumnDisplayMode,
+    SortOrder,
+    StartEditAction,
+    StateStoringType,
+    SummaryType,
+    ToolbarItemLocation,
+};
+
+export type ColumnResizingMode = 'nextColumn' | 'widget';
```
Comment (AnnaKondratova) [inline js/ui/data_grid.d.ts]:
```suggestion
export type ColumnResizeMode = 'nextColumn' | 'widget';
```

File: js/ui/form.d.ts [hunk 19]
```diff
@@ -48,14 +48,21 @@ import Widget, {
...
     Mode,
-    FormLabelLocation,
-    FormItemEditorType,
-    FormItemType,
-    FormLabelMode,
-} from '../types/enums';
+    VerticalAlignment,
+};
+
+export type FormEditor = 'dxAutocomplete' | 'dxCalendar' | 'dxCheckBox' | 'dxColorBox' | 'dxDateBox' | 'dxDropDownBox' | 'dxHtmlEditor' | 'dxLookup' | 'dxNumberBox' | 'dxRadioGroup' | 'dxRangeSlider' | 'dxSelectBox' | 'dxSlider' | 'dxSwitch' | 'dxTagBox' | 'dxTextArea' | 'dxTextBox';
```
Comment (AnnaKondratova) [inline js/ui/form.d.ts]:
```suggestion
export type FormWidget = 'dxAutocomplete' | 'dxCalendar' | 'dxCheckBox' | 'dxColorBox' | 'dxDateBox' | 'dxDropDownBox' | 'dxHtmlEditor' | 'dxLookup' | 'dxNumberBox' | 'dxRadioGroup' | 'dxRangeSlider' | 'dxSelectBox' | 'dxSlider' | 'dxSwitch' | 'dxTagBox' | 'dxTextArea' | 'dxTextBox';
```

File: js/ui/form.d.ts [hunk 20]
```diff
@@ -48,14 +48,21 @@ import Widget, {
...
+export type FormItem = 'empty' | 'group' | 'simple' | 'tabbed' | 'button';
```
Comment (AnnaKondratova) [inline js/ui/form.d.ts]:
```suggestion
export type FormItemType = 'empty' | 'group' | 'simple' | 'tabbed' | 'button';
```

File: js/ui/map.d.ts [hunk 21]
```diff
@@ -13,11 +13,9 @@ import Widget, {
     WidgetOptions,
 } from './widget/ui.widget';
 
-import {
-  GeoMapType,
-  GeoMapProvider,
-  GeoMapRouteMode,
-} from '../types/enums';
+export type MapProvider = 'bing' | 'google' | 'googleStatic';
+export type RoutingMode = 'driving' | 'walking';
```
Comment (AnnaKondratova) [inline js/ui/map.d.ts]:
```suggestion
export type RouteMode = 'driving' | 'walking';
```

File: js/ui/pivot_grid.d.ts [hunk 22]
```diff
@@ -23,15 +23,27 @@ import Widget, {
...
+export {
+    ChangesApplyMode,
+    FieldChooserLayout,
+    Mode,
+    ScrollMode,
+    StateStoringType,
+};
+
+export type PivotGridDataFieldArea = 'column' | 'row';
+export type PivotGridRowHeadersLayout = 'standard' | 'tree';
```
Comment (AnnaKondratova) [inline js/ui/pivot_grid.d.ts]:
```suggestion
export type PivotGridRowHeaderLayout = 'standard' | 'tree';
```

File: js/ui/pivot_grid.d.ts [hunk 23]
```diff
@@ -23,15 +23,27 @@ import Widget, {
...
+export type PivotGridTotalsDisplayMode = 'both' | 'columns' | 'none' | 'rows';
```
Comment (AnnaKondratova) [inline js/ui/pivot_grid.d.ts]:
```suggestion
export type PivotGridTotalDisplayMode = 'both' | 'columns' | 'none' | 'rows';
```

File: js/ui/scheduler.d.ts [hunk 24]
```diff
@@ -56,6 +52,17 @@ interface TargetedAppointmentInfo {
...
 
+export {
+    FirstDayOfWeek,
+    Orientation,
+    ScrollMode,
+};
+
+export type AllDayPanelMode = 'all' | 'allDay' | 'hidden';
+export type CellAppointmentsLimit = 'auto' | 'unlimited';
+export type RecurrenceEditingMode = 'dialog' | 'occurrence' | 'series';
```
Comment (AnnaKondratova) [inline js/ui/scheduler.d.ts]:
```suggestion
export type RecurrenceEditMode = 'dialog' | 'occurrence' | 'series';
```

File: js/viz/circular_gauge.d.ts [hunk 25]
```diff
@@ -20,10 +20,8 @@ import {
     TooltipInfo,
 } from './gauges/base_gauge';
 
-import {
-    CircularGaugeElementOrientation,
-    GaugeOverlappingBehavior,
-} from '../types/enums';
+export type CircularGaugeElementOrientation = 'center' | 'inside' | 'outside';
+export type CircularGaugeLabelOverlapping = 'first' | 'last';
```
Comment (AnnaKondratova) [inline js/viz/circular_gauge.d.ts]:
```suggestion
export type CircularGaugeLabelOverlapMode = 'first' | 'last';
```

File: js/viz/funnel.d.ts [hunk 26]
```diff
@@ -35,18 +35,33 @@ import BaseWidget, {
...
+    HorizontalEdge,
+    LabelPosition,
+    Palette,
+    PaletteExtensionMode,
+    TextOverflow,
+    WordWrap,
+};
+
+export type FunnelAlgorithm = 'dynamicHeight' | 'dynamicSlope';
+export type FunnelLabelOverlapping = 'hide' | 'none' | 'shift';
```
Comment (AnnaKondratova) [inline js/viz/funnel.d.ts]:
```suggestion
export type FunnelLabelOverlapMode = 'hide' | 'none' | 'shift';
```

File: js/viz/pie_chart.d.ts [hunk 27]
```diff
@@ -45,21 +45,32 @@ import {
...
+    HatchingDirection,
+    LabelPosition,
+    Palette,
+    TextOverflow,
+    WordWrap,
+};
+
+export type PieChartAnnotationLocation = 'center' | 'edge';
+export type PieChartLegendHoverMode = 'none' | 'allArgumentPoints';
+export type PieChartResolveLabelOverlapping = 'hide' | 'none' | 'shift';
```
Comment (AnnaKondratova) [inline js/viz/pie_chart.d.ts]:
```suggestion
export type PieChartResolveLabelOverlapMode = 'hide' | 'none' | 'shift';
```

File: js/viz/pie_chart.d.ts [hunk 28]
```diff
@@ -45,21 +45,32 @@ import {
...
+export type PieChartSegmentsDirection = 'anticlockwise' | 'clockwise';
```
Comment (AnnaKondratova) [inline js/viz/pie_chart.d.ts]:
```suggestion
export type PieChartSegmentDirection = 'anticlockwise' | 'clockwise';
```

File: js/common/grids.d.ts (hunk 3 above)
Comment (IlyaKhD) [reply js/common/grids.d.ts]: We agreed to use `Predefined` word in such cases

File: js/common/grids.d.ts (hunk 12 above)
Comment (IlyaKhD) [reply js/common/grids.d.ts]: done

File: js/common/grids.d.ts (hunk 11 above)
Comment (IlyaKhD) [reply js/common/grids.d.ts]: done

File: js/common/charts.d.ts (hunk 14 above)
Comment (IlyaKhD) [reply js/common/charts.d.ts]: done

File: js/common/charts.d.ts (hunk 15 above)
Comment (IlyaKhD) [reply js/common/charts.d.ts]: done

File: js/common/charts.d.ts (hunk 16 above)
Comment (IlyaKhD) [reply js/common/charts.d.ts]: done

File: js/common/charts.d.ts (hunk 17 above)
Comment (IlyaKhD) [reply js/common/charts.d.ts]: done

File: js/ui/data_grid.d.ts (hunk 18 above)
Comment (IlyaKhD) [reply js/ui/data_grid.d.ts]: done

File: js/ui/form.d.ts (hunk 20 above)
Comment (IlyaKhD) [reply js/ui/form.d.ts]: done

File: js/ui/map.d.ts (hunk 21 above)
Comment (IlyaKhD) [reply js/ui/map.d.ts]: done

File: js/ui/pivot_grid.d.ts (hunk 22 above)
Comment (IlyaKhD) [reply js/ui/pivot_grid.d.ts]: done

File: js/ui/scheduler.d.ts (hunk 24 above)
Comment (IlyaKhD) [reply js/ui/scheduler.d.ts]: done

File: js/viz/circular_gauge.d.ts (hunk 25 above)
Comment (IlyaKhD) [reply js/viz/circular_gauge.d.ts]: done

File: js/viz/funnel.d.ts (hunk 26 above)
Comment (IlyaKhD) [reply js/viz/funnel.d.ts]: done

File: js/viz/pie_chart.d.ts (hunk 28 above)
Comment (IlyaKhD) [reply js/viz/pie_chart.d.ts]: done

File: js/viz/pie_chart.d.ts (hunk 27 above)
Comment (IlyaKhD) [reply js/viz/pie_chart.d.ts]: Renamed it to `PieChartLabelOverlap` to keep consistent with others of the kind

File: js/animation/position.d.ts (hunk 7 above)
Comment (IlyaKhD) [reply js/animation/position.d.ts:12]: Decided to keep this name

File: js/animation/position.d.ts (hunk 8 above)
Comment (IlyaKhD) [reply js/animation/position.d.ts:14]: Decided to keep this name

File: js/common/charts.d.ts (hunk 10 above)
Comment (IlyaKhD) [reply js/common/charts.d.ts]: done

File: js/common/grids.d.ts (hunk 13 above)
Comment (IlyaKhD) [reply js/common/grids.d.ts]: done

File: js/ui/form.d.ts (hunk 19 above)
Comment (IlyaKhD) [reply js/ui/form.d.ts]: done

File: js/ui/pivot_grid.d.ts (hunk 23 above)
Comment (IlyaKhD) [reply js/ui/pivot_grid.d.ts]: done

**Review by OlegKipchatov — DISMISSED** [review]

**Review by OlegKipchatov — APPROVED** [review]

**Pull request merged**